### PR TITLE
Scope

### LINKED ISSUE #5: Arbitrary value can be set to default scope
```
az>> %%st
defaulting: st
az: error: argument _command_package: invalid choice: 'st' (choose from 'network', 'appservice', 'batch', 'keyvault', 'vmss', 'sql', 'storage', 'vm', 'iot', 'acs', 'resource', 'policy', 'find', 'container', 'snapshot', 'group', 'provider', 'disk', 'redis', 'documentdb', 'ad', 'cloud', 'component', 'login', 'lock', 'role', 'account', 'feedback', 'logout', 'image', 'feature', 'tag', 'configure')
usage: az [-h] [--output {json,tsv,table,jsonc}] [--verbose] [--debug]
          [--query JMESPATH]
          {network,appservice,batch,keyvault,vmss,sql,storage,vm,iot,acs,resource,policy,find,container,snapshot,group,provider,disk,redis,documentdb,ad,cloud,component,login,lock,role,account,feedback,logout,image,feature,tag,configure}
          ...
```

`st` is not a valid subcommand, nevertheless it is set to be a scope.

### LINKED ISSUE #6: After a scope is set redundant output is print.
```
az>> %% vm
defaulting: vm
az vm: error: the following arguments are required: subcommand
usage: az vm [-h]
             {availability-set,show,user,nic,image,boot-diagnostics,restart,diagnostics,redeploy,extension,generalize,unmanaged-disk,open-port,convert,list-sizes,wait,create,list,delete,disk,stop,resize,get-instance-view,capture,list-ip-addresses,list-vm-resize-options,deallocate,start,encryption,format-secret,update,list-usage}
             ...
```

The user sets the scope to vm. The following error message and brief help are redundant.

### PR DESCRIPTION
**Pull request by oakeyc**
fixes #5, closes #6, issue #34

### CODE REVIEW

File: MANIFEST.in [hunk 1]
```diff
@@ -0,0 +1 @@
+include *.rst
```
Comment (derekbekoe) [inline MANIFEST.in]: This change is included in another PR.

File: azclishell/app.py [hunk 2]
```diff
@@ -460,11 +459,25 @@ def handle_jmespath_query(self, text, continue_flag):
...
+            else:
+                tree_val = value
+
+            if in_tree(self.completer.command_tree, tree_val):
+                self.handle_scoping(value)
+                print("defaulting: " + value)
+                cmd = cmd.replace(SELECT_SYMBOL['default'], '')
+                telemetry.track_ssg('default command', value)
+            else:
+                print("not a valid value")
```
Comment (derekbekoe) [inline azclishell/app.py]: Is this for if the scope specified is not valid?
If so, the error message should say so.

File: azclishell/app.py [hunk 3]
```diff
@@ -460,11 +459,25 @@ def handle_jmespath_query(self, text, continue_flag):
...
+            if self.default_command:
+                tree_val = self.default_command + " " + value
+            else:
+                tree_val = value
+
+            if in_tree(self.completer.command_tree, tree_val):
+                self.handle_scoping(value)
+                print("defaulting: " + value)
+                cmd = cmd.replace(SELECT_SYMBOL['default'], '')
+                telemetry.track_ssg('default command', value)
```
Comment (derekbekoe) [inline azclishell/app.py]: Telemetry key should be changed. 'scope command' maybe.

File: azclishell/app.py [hunk 4]
```diff
@@ -460,11 +459,25 @@ def handle_jmespath_query(self, text, continue_flag):
...
+                value = default
+
+            if self.default_command:
+                tree_val = self.default_command + " " + value
+            else:
+                tree_val = value
+
+            if in_tree(self.completer.command_tree, tree_val):
+                self.handle_scoping(value)
+                print("defaulting: " + value)
```
Comment (derekbekoe) [inline azclishell/app.py]: Suggest: `Scope set: `

File: azclishell/az_completer.py [hunk 5]
```diff
@@ -125,18 +125,18 @@ def get_completions(self, document, complete_event):
...
         text = reformat_cmd(text)
-
         if text.split():
-            for comp in self.gen_cmd_and_param_completions(text):
+            for comp in sorted(
+                    list(self.gen_cmd_and_param_completions(text)), key=lambda a: a.text):
                 yield comp
 
-        for cmd in self.gen_cmd_completions(text):
+        for cmd in sorted(list(self.gen_cmd_completions(text)), key=lambda a: a.text):
```
Comment (derekbekoe) [inline azclishell/az_completer.py]: I saw this change in another PR..

**Review by derekbekoe — CHANGES_REQUESTED** [review]
Looks like some changes exist in multiple PRs.

File: MANIFEST.in (hunk 1 above)
Comment (oakeyc) [reply MANIFEST.in]: That's because of the way I made the branches

File: azclishell/az_completer.py (hunk 5 above)
Comment (oakeyc) [reply azclishell/az_completer.py]: That's because of the way I made the branches

**Comment by oakeyc** [comment]
changes made

**Review by derekbekoe — APPROVED** [review]

**Pull request merged**